jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/WebSocketWriteTimeoutException.java

```diff
@@ -0,0 +1,27 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.websocket.core;
+
+public class WebSocketWriteTimeoutException extends WebSocketTimeoutException
```

Comment:
why do we need this?

Reply:
I added this in so we can detect this exception in `WebSocketChannel.processConnectionError()` and set a status of `NO_CLOSE` so we don't attempt to send a close frame as we know we are failing the flusher so it wont be able to send

Reply:
OK - a good use of an exception type... something that is explicitly handled.